docs/usage/getting-started.md

```diff
@@ -89,6 +89,24 @@ At this point, **all your translation files** will be loaded on the client side,
 
 We also recommend to always use either [loadTranslations](./server-side-loading.md) or [clientNamespaces](./client-side-loading.md) or both in a page, else you will load all the namespaces on your application for every page.
 
+### Overriding the initial locale
+
+By default, the i18n instance is initialized with the locale from Next.js. If you want to override this (for example, with a locale from a cookie, header, or some other method), pass that locale as the pageProp `__ni18n_locale__`.
+
+```typescript
+// pages/index.ts
+
+export const getServerSideProps = async ({ req }) => {
+  const locale = req.cookies['MY_LOCALE_COOKIE']
+  return {
+    props: {
+      __ni18n_locale__: locale,
+      ...(await loadTranslations(ni18nConfig, locale, ['server-namespace'])),
```

Comment:
What do you think about adding this as a prop to `loadTranslations`? So we can use it like described below? This way we would be easier to typecheck everything as well.

```ts
await loadTranslations(
  ni18nConfig, 
  locale, 
  ['server-namespace'], 
  { overrideRouterLocale: true }
)
```

and then inside `loadTranslations` we can simply return 

```ts
  return {
    __ni18n_server__: {
      resources: store,
      ns: namespaces,
      ...overrideRouterLocale && { locale }
    },
  }
```

Reply:
Ah, and then the locale from `__ni18n_server__` automatically overrides the locale in `WithI18Next`. That is a more ergonomic solution, I'll make some updates. Thanks!

Reply:
Would it ever make sense to not override the router locale if you're using `loadTranslations` ? Shouldn't they always be in sync?

Reply:
Possibly, idk, it has been a while since I had to use translations. If they don't work without the override, then we might as well try just always overriding it.

Reply:
Updated, thank you for the feedback!